src/Gonimo/Server/Handlers.hs

```diff
@@ -45,6 +48,18 @@ createClient mUserAgent = do
       }
 
 
+-- You may test this part of the API this way:
+-- $ curl --request POST http://localhost:8081/funnyusername
+createFunnyUserName :: ServerConstraint r => Eff r Text
+createFunnyUserName = do
+  let scaffold     = [FunnyPrefix1, FunnyPrefix2, FunnyCharacter, FunnySuffix]
+      fetch t      = Db.selectList [FunnyWordWordType ==. t] []
+      word r       = funnyWordWord (entityVal r)
+  funnyWordPools  <- fmap (fmap (fmap word)) $
+                     runDb $ mapM fetch scaffold
+  funnyWords      <- runRandom $ randomLs funnyWordPools
```

Comment:
Instead of randomLs we really should already only pick the words from the database we want, instead of pulling all words to memory and doing a linear list search then. Constant memory & constant runtime vs. linear memory and linear runtime.

If needed we can add more effects to Gonimo.Database.Effects.